base/sharedarray.jl

```diff
@@ -249,7 +249,7 @@ function rand!{T}(S::SharedArray{T})
 end
 
 function randn!(S::SharedArray)
-    f = S->map!(x->randn, S.loc_subarr_1d)
+    f = S->map!(x->randn(), S.loc_subarr_1d)
```

Comment:
The discussion in #9836 suggests that converting the result of `randn()` to other floating point types is not quite right. If we're not defining `randn!(::Array{Float32})` because of this, then I don't think `randn!(::SharedArray{Float32})` should work either.